apps/molgenis-components/src/client/client.ts

```diff
@@ -402,12 +402,21 @@ const columnNames = (
   schemaName: string,
   tableName: string,
   metaData: ISchemaMetaData,
-  expandLevel: number
+  //allows expansion of ref fields to add their next layer of details. 
+  expandLevel: number,
+  //default is rootLevel listing of columnNames
+  rootLevel: boolean = true
```

Comment:
param rootLevel  is only used in for recursion stop, function should have (non recursive) public api without rootlevel param that calls private (recursive) function where the rootLevel is set to true only on the initial kickoff